pkg/states/state_manager.go

```diff
@@ -0,0 +1,77 @@
+package states
+
+import "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+
+func Active(userSignup *v1alpha1.UserSignup) bool {
+	return Approved(userSignup) &&
+		!VerificationRequired(userSignup) &&
+		!Deactivated(userSignup)
+}
+
+func Approved(userSignup *v1alpha1.UserSignup) bool {
+	return contains(userSignup.Spec.States, v1alpha1.UserSignupStateApproved)
+}
+
+func SetApproved(userSignup *v1alpha1.UserSignup, val bool) {
```

Comment:
I'm slightly missing the purpose of the boolean parameter? Why should I want to call `SetApproved(userSignup, false)` that doesn't make sense to me - if I want to approve a user, then I just call `SetApproved(usersignup)` or just `Approve(userSignup)`

or is there any use-case when we would need to call `SetApproved(userSignup, false)`?

Reply:
We could probably omit the parameter in the case of `SetApproved`, however in other cases it's necessary - for example calling `SetVerificationRequired(userSignup, false)` once the user successfully verifies.  On one hand I'm open to removing the parameter in this case, however on the other hand it seems that consistency is important also.  I guess I'm sitting on the fence here ;)

Reply:
fair enough - let's keep it as is it

Reply:
And here too. We might change this when we decide what we want to have as a definition of the desired "active/approved" state.